src/foam/core/stdlib.js

```diff
@@ -827,7 +827,23 @@ foam.LIB({
 
       return newObj;
     },
-    function equals(a, b) { return a === b; },
+    function keys(o) { return Object.keys(o); },
+    function equals(a, b) {
+      if ( foam.Object.is(a, b) ) return true;
+      if ( ! foam.Object.isInstance(a)
+        || ! foam.Object.isInstance(b) ) return false;
+      if ( this.keys(a).length !== this.keys(b).length ) return false;
+
+      for ( var key in a ) {
```

Comment:
This won't check enumerable properties. I'm not sure if we want to check those or not anyway though. If we did, we could use [getOwnPropertyNames](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames).

Just another note in case it matters, by checking `hasOwnProperty` in the loop, we're only going to be looking at properties owned directly by the objects. Which means we won't be checking any properties that are inherited through the prototype chain. Again, I'm not sure whether we want to or not, but I wanted to point it out in case that matters.

Reply:
Good point. @kgrgreer, do you think we need to account for non-enumerable properties when comparing objects using `foam.Object.equals`?

Reply:
We're really only using this for map-like objects, so I think this is fine. Following prototype inheritance could involve comparing the whole system.